Create DOD (Definition of Done) (Issue/2688) (#2703)
porsche-design-system/porsche-design-system: packages/storefront/src/pages/must-know/definition-of-done.md

# Definition Of Done

We value high quality and transparency.  
Thus we consider a task or ticket only done when all of the following criteria are fulfilled as our **Definition of
Done** (DOD).

<TableOfContents></TableOfContents>

## Current Criteria

> 

Reviewer: Maybe we should also add things like, code is written with: 
- Semantic HTML
- Smallest possible DOM hierarchy
- Modern CSS techniques producing as less CSS as possible
- In general having everything properly typed, abstracted, tested and maintainable
…

- Unit tests (in node/jsdom)
- End-to-End Tests (E2E, via puppeteer in Chrome)
  - relevant functionality
  - a11y compliance (a11y tree snapshots)

> - Visual Regression Tests (VRT, via puppeteer in Chrome and playwright in Desktop Safari)

Reviewer: ```suggestion
- Visual Regression Tests (VRT, via Puppeteer in Chromium (Chrome) and Playwright in WebKit (Safari))
```

  - different prop configurations per component
  - :hover, :focus and :hover:focus states
  - high contrast mode
  - 200% text zoom support
- Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)

> - SSR support for Next.js and Remix (unit tested, and visually tested via puppeteer in Chrome with disabled JavaScript)

Reviewer: ```suggestion
- SSR support for Next.js and Remix (unit tested, and visually tested via Puppeteer  with disabled JavaScript)
```

- Smoke tests (if adding new assets to our CDN)
- Security vulnerability scan passes (CodeQL of GitHub)
- Cross Browser Support (last 2 stable versions of Chrome, Edge, Firefox, iOS and macOS Safari)
- Documentation updated
- Changelog updated
- Code reviewed (via pull request)

> 

Reviewer 2: Should we mention UXPin as well?

# Definition Of Done

We value high quality and transparency.  
Thus we consider a task or ticket only done when all of the following criteria are fulfilled as our **Definition of
Done** (DOD).

<TableOfContents></TableOfContents>

## Current Criteria

- Respecting Porsche Brand Identity
- Code is written with
  - Semantic HTML
  - Smallest possible DOM hierarchy
  - Modern CSS techniques producing as few CSS as possible
  - Performance at runtime
  - Proper typing, abstraction, reusability and maintainability

> - Code is tested with
>   - Unit tests (in node/jsdom)
>   - End-to-End Tests (E2E, via puppeteer in Chrome)
>   - Visual Regression Tests (VRT, via Puppeteer in Chromium and Playwright in WebKit browser engines)
>   - Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)
>   - Smoke tests (if adding new assets to our CDN)

Reviewer: ```suggestion
- Code is tested with
  - Unit tests (in node/jsdom)
  - End-to-End Tests (E2E, via puppeteer in Chrome)
  - Accessibility Tests (A11y, via Axe Core)
  - Visual Regression Tests (VRT, via Puppeteer in Chromium and Playwright in WebKit browser engines)
  - Unit tests for/with jsdom-polyfill (if modern browser APIs are used that are not available in jsdom)
  - Smoke tests (if adding new assets to our CDN)
```

> - Accessibility compliance (WCAG 2.1 level AA)
>   - High Contrast Mode
>   - 200% Text Zoom Support

Reviewer: ```suggestion
- Accessibility compliance
  - WCAG 2.1 level AA
  - High Contrast Mode
  - 200% Text Zoom Support
```
To make clear, that high contrast + text-zoom is something additional.
